gapis/resolve/dependencygraph2/graph_visualization/graph_visualization.go

```diff
@@ -16,17 +16,78 @@ package graph_visualization
 
 import (
 	"context"
+	"fmt"
 	"github.com/google/gapid/core/math/interval"
 	"github.com/google/gapid/gapis/api"
 	"github.com/google/gapid/gapis/capture"
 	"github.com/google/gapid/gapis/resolve/dependencygraph2"
 )
 
-func GetGraphVisualizationFromCapture(ctx context.Context, p *capture.Capture) ([]byte, error) {
+func createGraphFromDependencyGraph(dependencyGraph dependencygraph2.DependencyGraph) (*graph, error) {
+
+	numberOfNodes := dependencyGraph.NumNodes()
+	currentGraph := createGraph(0)
+	currentHierarchy := &api.Hierarchy{}
+	previousNode := &node{}
+
+	for i := 0; i < numberOfNodes; i++ {
+		dependencyNode := dependencyGraph.GetNode(dependencygraph2.NodeID(i))
+		if cmdNode, ok := dependencyNode.(dependencygraph2.CmdNode); ok {
+			cmdNodeId := cmdNode.Index[0]
+			command := dependencyGraph.GetCommand(api.CmdID(cmdNodeId))
+			commandName := command.CmdName()
+
+			if graphVisualizationAPI, ok := command.API().(api.GraphVisualizationAPI); ok {
+				label := graphVisualizationAPI.GetCommandLabel(currentHierarchy, command)
+				label += fmt.Sprintf("%s%d", commandName, cmdNodeId)
+				label += graphVisualizationAPI.GetSubCommandLabel(cmdNode.Index)
+				label += "\""
+
+				attributes := fmt.Sprintf("\"%v\"", command.CmdParams())
+
+				newNode := currentGraph.getNewNode(i, label)
+				newNode.name = commandName
+				newNode.attributes = attributes
+				currentGraph.addNode(newNode)
 
+				currentNode := currentGraph.nodeIdToNode[i]
+				if graphVisualizationAPI.IsInsideRenderScope(previousNode.name) {
+					if graphVisualizationAPI.IsInsideRenderScope(currentNode.name) {
+						if graphVisualizationAPI.AreLabelsInRenderLevel(previousNode.label, currentNode.label) {
+							currentGraph.addEdgeBetweenNodes(currentNode, previousNode)
+						}
+					}
+				}
+				if graphVisualizationAPI.IsInsideRenderScope(currentNode.name) {
+					previousNode = currentNode
+				}
```

Comment:
Yes, in a later commit it is removed. I was considering put that functions in the interface because in a later commit ,label is not longer a string, but I think it would be better just consider in that commit.
So I removed all code in regard to adding a chain of edges through the render pass.